post installation script update

mmattel opened the pull request:
This is an update of the post installation script.

Following new:
- a bit more description
- automation if you want to upgrade an existing installation
- asks for the tar file
- renames the original directory including a timestamp
- creates a new directory with the correct name
- extracts correctly or stops in case of an error during extracting

settermjd commented on modules/ROOT/examples/installation/post-installation-steps.sh:
```diff
@@ -1,11 +1,25 @@
+
+ocname='owncloud'
+ocroot='/var/www/'
+ocpath=$ocroot$ocname
```
How about quoting this and the following variables as an extra sanity check? Also, while it's a little more work, okay one character per variable, how about changing `ocroot` to `'/var/www'`, and then initialising the others with a forward slash between the variables, e.g., `ocpath=$ocroot/$ocname`. It could just be me, but I find compound variables easier to read when declared that way. Plus, I can't say for certain, but I thought it was an informal convention in shell scripts.

settermjd commented on modules/ROOT/examples/installation/post-installation-steps.sh:
```diff
@@ -17,15 +31,24 @@ rootuser='root'
 
 echo
+read -p "Do you want to upgrade an existing installation (y/N)? " -r -e answer
+if echo "$answer" | grep -iq "^y"; then
```
You could simplify the if/else to `(echo "$answer" | grep -iq "^y") && do_upgrade="y" || do_upgrade="n"` instead, as it's just a simple test + assignment.

settermjd commented on modules/ROOT/examples/installation/post-installation-steps.sh:
```diff
@@ -47,6 +70,23 @@ else
 
+# check if to upgrade an existing installation
+if [ "$do_upgrade" = "y" ]; then
+  read -p "Please specify the file to extract with full path: " -r -e answer
```
I see that the convention you're using is to use `$answer` for storing user input, which makes perfect sense, but would it not make more sense to name the variable based on it's intended use?

mmattel replied:
it is just the given answer.
more important is the variable having the name which highlights its itention.

settermjd replied:
Fair point. It was worth raising.

mmattel commented on modules/ROOT/examples/installation/post-installation-steps.sh:
```diff
@@ -83,7 +124,8 @@ else
     printf "ln $ocapps_external \n"
-    ln -sfn $linkapps-external $ocapps_external
+	mkdir -p $linkapps_external
+    ln -sfn $linkapps_external $ocapps_external
```
ups, this was not intended, it is a tab instead blanks

mmattel commented on modules/ROOT/examples/installation/post-installation-steps.sh:
```diff
@@ -47,6 +70,23 @@ else
+    mv $ocpath $oldocpath
+  fi
+  mkdir -p $ocpath
+  if ! tar xvf "$answer" -C $ocpath --strip-components=1 ; then
```
Tried it now for hours. Atm where I put any kind of brackets around, it will no longer work...

mmattel commented:
Implemented recommendations.
Added one check before chmod to check, if files are present, else chmod would return an error. Identified when playing around with the tar command (if tar would succeed but no files have been extracted... very rare but possible)

mmattel commented:
I just adopted the script´s path to the new administration_manual path instead of root 😄

settermjd commented on modules/administration_manual/examples/installation/post-installation-steps.sh:
```diff
@@ -126,6 +164,7 @@ if [ "$chmdir" = "y" ] && [ "$uselinks" = "y" ]; then
   find -L ${ocapps_external}/ -type d -print0 | xargs -0 chmod 0750
 fi
 
+#chown
```
Can you make this more descriptive, pls?

settermjd replied:
Am I missing something? It doesn't seem to be visible.

settermjd replied:
Unless mine's out of date, I don't see it, still.

mmattel commented:
I rebased and pushed. Now it is here 😄 
Sorry for the confusion...
